python/ray/data/dataset.py

```diff
@@ -236,14 +239,15 @@ def __init__(
         usage_lib.record_library_usage("dataset")  # Legacy telemetry name.
 
         self._plan = plan
-        self._set_uuid(uuid4().hex)
         self._logical_plan = logical_plan
         if logical_plan is not None:
             self._plan.link_logical_plan(logical_plan)
 
         # Handle to currently running executor for this dataset.
         self._current_executor: Optional["Executor"] = None
         self._write_ds = None
+        self._uuid = None
+        self._set_uuid(get_dataset_id_from_stats_actor())
```

Comment:
is there any chance `get_dataset_id_from_stats_actor()` would fail? Let's add a fallback if `get_dataset_id_from_stats_actor()` is failed, then we still set uuid with `uuid4().hex`. All in all, we should make this as a best effort, and not fail on critical path.